jaxrs-api/src/test/java/javax/ws/rs/core/MediaTypeTest.java

```diff
@@ -58,4 +60,110 @@ public void testNullConstructorValues() {
         actual = new MediaType(null, null, (String) null);
         assertEquals(MediaType.WILDCARD_TYPE, actual);
     }
+
+    /**
+     * Test that a wildcard type {@link MediaType} value with explicit subtype is not compatible with
+     * a non-wildcard type {@link MediaType} value with a different subtype.
+     */
+    @Test
+    public void testMediaTypeWithWildcardTypeNotCompatibleWhenSubtypeDifferent() {
+        MediaType wildcard = new MediaType(MediaType.MEDIA_TYPE_WILDCARD, "json");
+
+        assertFalse(wildcard.isCompatible(MediaType.APPLICATION_OCTET_STREAM_TYPE));
+        assertFalse(MediaType.APPLICATION_OCTET_STREAM_TYPE.isCompatible(wildcard));
+    }
+
+    /**
+     * Test that a wildcard type {@link MediaType} value with explicit subtype is compatible with
+     * a non-wildcard type {@link MediaType} value with the same explicit subtype.
+     */
+    @Test
+    public void testMediaTypeWithWildcardTypeCompatibleWhenSubtypeMatches() {
+        MediaType wildcard = new MediaType(MediaType.MEDIA_TYPE_WILDCARD, "json");
+
+        assertTrue(wildcard.isCompatible(MediaType.APPLICATION_JSON_TYPE));
+        assertTrue(MediaType.APPLICATION_JSON_TYPE.isCompatible(wildcard));
+    }
+
+    /**
+     * Test that a wildcard type {@link MediaType} value with explicit subtype is not compatible with
```

Comment:
This comment tells a different story than this tests actual source code.

Reply:
@mszabo-wikia Did I misunderstand the JavaDoc, or is another commit in preparation? :-)

Reply:
@mkarg I took a look and this was a copy-paste of the first test case's description by accident, sorry about the confusion :)